linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/ClassEmitter.scala

```diff
@@ -192,63 +192,66 @@ private[emitter] final class ClassEmitter(sjsGen: SJSGen) {
     implicit val pos = tree.pos
 
     val className = tree.name.name
-    val isJSClass = tree.kind.isJSClass
-    val typeVar =
-      if (isJSClass) fileLevelVar("b", genName(className))
-      else globalVar("c", className)
-
-    def makeInheritableCtorDef(ctorToMimic: js.Tree, field: String) = {
-      js.Block(
-        js.DocComment("@constructor"),
-        if (isJSClass) globalVarDef(field, className, js.Function(false, Nil, js.Skip()))
-        else globalFunctionDef(field, className, Nil, js.Skip()),
-        globalVar(field, className).prototype := ctorToMimic.prototype
-      )
-    }
 
-    val ctorFunWithGlobals =
-      if (!isJSClass) genJSConstructorFun(tree, initToInline)
-      else genConstructorFunForJSClass(tree)
+    def assignConstructorProperty(ctorVar: js.Tree): js.Tree =
+      genIdentBracketSelect(ctorVar.prototype, "constructor") := ctorVar
```

Comment:
Consider inlining this at the two call sites. This is only line that is still "factored out", and the factoring out makes it actually harder to understand what's going on below. I had to come back to this definition and mentally inline it at the call sites to make sense of them.

Reply:
I wasn't sure about this one myself :) Removed.